sympy/simplify/simplify.py

```diff
@@ -33,6 +33,38 @@
 import mpmath
 
 
+def separable(expr, symbols=[]):
+    """Return the subset of symbols which appear in expr that
+    test as being positively separable from other symbols in expr
+    and those that do not test as being unseparable.
+
+    If `f(x, y)` is separable in `x` then `f(x, y).diff(x)/f(x, y)`
+    should be constant wrt y.
+
+    Examples
+    ========
+
+    >>> from sympy.simplify.simplify import separable
+    >>> from sympy import cos
+    >>> from sympy.abc import x, y
+    >>> eq = cos(x + y) + cos(x - y)
+    >>> separable(eq)
+    ({x, y}, set())
+    """
+    s = []
+    u = set()
+    free = expr.free_symbols
+    symbols = set(symbols) or free
+    present = free & set(symbols)
```

Comment:
Since symbols is not use below we can combine the two lines above to
```
present = free & (set(symbols) or free)
```
We can simplify that to
```
present = set(free)
if symbols:
    present = present.intersection(symbols)
```
which is more efficient.